python/lsst/daf/butler/registry/sql_registry.py

```diff
@@ -289,7 +300,9 @@ def copy(self, defaults: RegistryDefaults | None = None) -> SqlRegistry:
             # No need to copy, because `RegistryDefaults` is immutable; we
             # effectively copy on write.
             defaults = self.defaults
-        return type(self)(self._db, defaults, self._managers)
+        result = SqlRegistry(self._db, defaults, self._managers)
+        result.dimension_record_cache.load_from(self.dimension_record_cache)
+        return result
```

Comment:
I hadn't actually realized the `CachingDimensionRecordStorage` instance was being shared between the `Butler` instances before; mostly I hadn't thought about what this sort of `Butler` construction does at all since until very recently it's mostly been for unit testing.

But I do think this is consistent with your plan for having different `Butler` instances each get their own cache when constructed by the factory, right?

Reply:
Yeah I think whatever happens to copy or share the cache happens right here, so this is fine structurally.  As long as nothing is depending on that behavior currently (which it probably isn't) I think it's fine to leave this as-is.